merlin/models/tf/models/base.py

```diff
@@ -1855,13 +1866,65 @@ def from_config(cls, config, custom_objects=None):
         if schema is not None:
             schema = schema_utils.tensorflow_metadata_json_to_schema(schema)
 
-        return cls(*layers, pre=pre, post=post, schema=schema)
+        model = cls(*layers, pre=pre, post=post, schema=schema)
+
+        inputs = model.get_sample_inputs(batch_size=batch_size)
+        if inputs:
+            model(inputs)
+
+        return model
+
+    def get_sample_inputs(self, batch_size=None):
+        batch_size = batch_size or 2
+        if self.input_schema is not None:
+            inputs = {}
+            for column in self.input_schema:
```

Comment:
Maybe the issue is schema filtering/propagation? If for example I change these lines to go through `self.schema` instead of `input_schema`, i.e.,
```
        if self.schema is not None:
            inputs = {}
            for column in self.schema:
````
and change the notebook to use `sub_schema` in the model by passing `sub_schema`: 
```
model = mm.Model(deep_dlrm_interaction, binary_task, schema=sub_schema)
```
I can get all tests to succeed.

Reply:
The test passing is not equivalent to it running correctly unfortunately. Using self.schema instead of self.inout_schema results in an exception being thrown inside the model from_config method where we call `model(inputs)`. This is because we end up passing the target `rating_binary` as an inout which the model does not accept. That exception is caught somewhere in TensorFlow/Keras, which seems to result in essentially skipping this model call, and reverting to the previous mechanism for building the model during load.

Reply:
It turns out that this new saving v3 lib in keras seems like it's only used during pickle and not a regular saved model that we do with model.save(). I've added a condition to apply this model build in from config based on this threadlocal that is set in there. 

https://github.com/NVIDIA-Merlin/models/pull/1040/commits/8e55be10195f4828c374b36dc40a0dc16d92c73d

Reply:
This means that most models will be pickleable. Although we only explictly test this in one test `test_pickle`. However, there is at least one kind of model that isn't pickleable (the one in define your own architecture notebook).

Reply:
This same saving_lib that is being used for pickle. is now the main saving lib in 2.12. So I think we'll have some more work to get that working reliably for all models.

Reply:
This seems like a good solution. I believe it's the same with the upstream keras models, i.e., most but not all keras models are pickable, but the recommended way is to use the `save()` method not pickle.

Reply:
It looks like `test_pickle` is failing after 8e55be1 and a916914 :(

Reply:
Looks like GPU CI was testing against tensorflow/keras 2.12 where the `saving_lib` moved from `keras.saving.experimental.saving_lib` to `keras.saving.saving_lib`. I've added an extra import for this and seems to work. This feels kind of brittle in the sense that this could break easily in future versions when the temporary `_SAVING_V3_ENABLED` variable is removed. Unless we have some way to detect if the model is being saved with this new keras format (used for pickle) vs the saved model format.

Reply:
Perhaps we can figure out if we'd like to try supporting the keras native format properly which does seem to have uncovered some bugs in various model layers, so from that perspective seems worrthwhile.